features/netsocket/UDPSocket.cpp

```diff
@@ -37,6 +32,13 @@ nsapi_protocol_t UDPSocket::get_proto()
     return NSAPI_UDP;
 }
 
+nsapi_error_t UDPSocket::connect(const SocketAddress &address)
+{
+    if (!address)
```

Comment:
POSIX permits a null address to undo the connection, so I think you can just take this check out.